arkouda/segarray.py

```diff
@@ -640,6 +647,8 @@ def append_single(self, x, prepend=False):
         SegArray
             Copy of original SegArray with values from x appended to each sub-array
         """
+        if self.dtype == str_:
+            raise TypeError("String elements are immutable and cannot accept a single value")
```

Comment:
I think we should be able to do this, but I think it's outside the scope of this PR. I'll open an issue to capture this

Reply:
Agreed. Thanks

Reply:
Captured in #2471